controller/configentry_controller.go

```diff
@@ -105,16 +110,31 @@ func (r *ConfigEntryController) ReconcileEntry(
 		// The object is being deleted
 		if containsString(configEntry.GetObjectMeta().Finalizers, FinalizerName) {
 			logger.Info("deletion event")
-			// Our finalizer is present, so we need to delete the config entry
-			// from consul.
-			_, err := r.ConsulClient.ConfigEntries().Delete(configEntry.ConsulKind(), configEntry.Name(), &capi.WriteOptions{
+			// Check to see if consul has config entry with the same name
+			entry, _, err := r.ConsulClient.ConfigEntries().Get(configEntry.ConsulKind(), configEntry.Name(), &capi.QueryOptions{
 				Namespace: r.consulNamespace(req.Namespace, configEntry.ConsulNamespaced()),
 			})
-			if err != nil {
-				return ctrl.Result{}, fmt.Errorf("deleting config entry from consul: %w", err)
-			}
-			logger.Info("deletion from Consul successful")
 
+			// Ignore the error where the config entry isn't found in Consul.
+			// It is indicative of desired state.
+			if err != nil && !isNotFoundErr(err) {
+				return ctrl.Result{}, fmt.Errorf("getting config entry from consul: %w", err)
+			} else if err == nil {
+				// Only delete the resource from Consul if it is owned by our datacenter.
+				if entry.GetMeta()[common.DatacenterKey] == r.DatacenterName {
+					// Our finalizer is present, so we need to delete the config entry
+					// from consul.
```

Comment:
Can probably drop this comment. Doesn't make much sense here since we're so deep into the deletion process here.